### DISCUSSION TITLE
Any luck with WorkerService and Console app/Service?

### DISCUSSION THREAD

**Discussion by jevgenijusmarinuskinas** [microsoft/ApplicationInsights-SnapshotCollector#13] [Q&A]
Hi,

I am unable to get snapshots with using non-ASP application, which is a Console app/Windows Service. We are using an AppInsights worker service, and initialize the SnapshotCollector in this fashion:

`services.AddApplicationInsightsTelemetryProcessor<SnapshotCollectorTelemetryProcessor>();
                services.AddSnapshotCollector(config =>
                {
                    config.IsEnabled = true;
                    config.IsEnabledInDeveloperMode = true;
                });`
  Snapshots are not coming to Azure App Insights, while telemetry is coming on exceptions. Is it not supported or there is some configuration issue?

**Comment by pharring** [comment]
Make sure you're using the latest preview of Microsoft.ApplicationInsights.SnapshotCollector (1.4.0-pre-2011.801 at time of writing) and also referencing Microsoft.ApplicationInsights.WorkerService.

Here's a complete example:

**Program.cs**
```csharp
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

namespace WorkerService3
{
    public class Program
    {
        public static void Main(string[] args)
        {
            CreateHostBuilder(args).Build().Run();
        }

        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureServices((hostContext, services) =>
                {
                    services.AddHostedService<Worker>();
                    services
                        .AddApplicationInsightsTelemetryWorkerService()
                        .AddSnapshotCollector(options => hostContext.Configuration.Bind("ApplicationInsights:SnapshotCollector", options));
                });
    }
}
```

**appsettings.json**
```json
{
  "Logging": {
    "LogLevel": {
      "Default": "Information",
      "Microsoft": "Warning",
      "Microsoft.Hosting.Lifetime": "Information"
    }
  },
  "ApplicationInsights": {
    "ConnectionString": "<your connection string>",
    "SnapshotCollector": {
      "IsEnabledInDeveloperMode": true
    }
  }
}
```
